runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -276,13 +277,11 @@ public class StreamingDataflowWorker {
   private final HotKeyLogger hotKeyLogger;
   // Periodic sender of debug information to the debug capture service.
   private final DebugCapture.@Nullable Manager debugCaptureManager;
-  private ScheduledExecutorService refreshWorkTimer;
-  private ScheduledExecutorService statusPageTimer;
-  private ScheduledExecutorService globalWorkerUpdatesTimer;
+  // Map from timer name to ScheduledExecutorService
+  private ConcurrentMap<String, ScheduledExecutorService> timersMap = new ConcurrentHashMap();
```

Comment:
I think that timersMap is a little confusing given the windmill Timer terminology.
How about just scheduledExecutors? or periodicFunctionExecutors?

Also as a nit, we're not using the key or using this concurrently and it makes things a little more complicated. I think you could change this to an ArrayList and avoid repeating the key. You can also only create the executor if needed.

Example:
```
if (...) {
  ScheduledExecutorService workUpdatesService = 
     executorSupplier.apply("GlobalWorkerUpdates");
  workUpdatesService.scheduleWithFixedDelay();
  scheduledExecutors.add(workUpdateService);
}
```

Reply:
Done.